### PR TITLE
[SYCL] Remove support of deprecated {121, 1.2.1, sycl-1.2.1, 2017} values for the sycl-std= argument

### PR DESCRIPTION
**Pull request by AndreiZibrov**
As a continuation of #14261 this PR makes impossible to call clang with -sycl-std=2017 option as well as with 121, 1.2.1, sycl-1.2.1 values. It also removes some tests which are available only with support of 2017/121

The rest of cleanup is to be continued with next submissions

### CODE REVIEW

File: clang/test/Driver/sycl-deprecated.cpp [hunk 1]
```diff
@@ -1,22 +0,0 @@
-/// Test for any deprecated options
-// RUN: %clangxx -fsycl-explicit-simd %s -### 2>&1 | FileCheck %s -DOPTION=-fsycl-explicit-simd
-// RUN: %clangxx -fno-sycl-explicit-simd %s -### 2>&1 | FileCheck %s -DOPTION=-fno-sycl-explicit-simd
-// RUN: %clangxx -fsycl -fsycl-link-huge-device-code %s -### 2>&1 | FileCheck %s -DOPTION=-fsycl-link-huge-device-code
-// RUN: %clangxx -fsycl -fno-sycl-link-huge-device-code %s -### 2>&1 | FileCheck %s -DOPTION=-fno-sycl-link-huge-device-code
-// RUN: %clangxx -fsycl -fsycl-use-bitcode %s -### 2>&1 | FileCheck %s -DOPTION=-fsycl-use-bitcode
-// RUN: %clangxx -fsycl -fno-sycl-use-bitcode %s -### 2>&1 | FileCheck %s -DOPTION=-fno-sycl-use-bitcode
-// CHECK: option '[[OPTION]]' is deprecated and will be removed in a future release
```
[the remark this answers is not in the document]
Comment (AndreiZibrov) [reply clang/test/Driver/sycl-deprecated.cpp:8]: Thank you. Done

**Review by dm-vodopyanov — COMMENTED** [review]
Also need to remove/adjust:
* https://github.com/intel/llvm/blob/46528f91dc23b3fa523c3650facbb1c6b279ef89/sycl/test-e2e/Sampler/basic-rw.cpp#L12
* https://github.com/intel/llvm/blob/46528f91dc23b3fa523c3650facbb1c6b279ef89/sycl/doc/design/CompilerAndRuntimeDesign.md?plain=1#L270
* https://github.com/intel/llvm/blob/46528f91dc23b3fa523c3650facbb1c6b279ef89/sycl/doc/UsersManual.md?plain=1#L136
* many tests in clang/test/SemaSYCL/ still uses removed values, as well as in in clang/test/Driver. Just grep for `-sycl-std` in the whole intel/llvm repo.

**Comment by dm-vodopyanov** [comment]
And I'm curious, how all these not removed tests are passing in pre-commit, if they still use removed values? Are we showing just some warning, and not an error, if user uses removed values for `-sycl-std` option?
Tagging @mdtoguchi

**Comment by AlexeySachkov** [comment]
> And I'm curious, how all these not removed tests are passing in pre-commit, if they still use removed values? Are we showing just some warning, and not an error, if user uses removed values for `-sycl-std` option? Tagging @mdtoguchi

`SemaSYCL` tests don't go through driver and therefore they most likely bypass argument validity check for the option. It then defaults to `SYCL_None`, but doesn't seem to cause any differences, i.e. they don't seem to test anything that is language-version-specific anymore.

**Comment by AndreiZibrov** [comment]
>many tests in clang/test/SemaSYCL/ still use removed values, as well as in in clang/test/Driver/ and clang/test/CodeGenSYCL/. Just grep for -sycl-std in the whole intel/llvm repo.

I'd be happy to know how many and which tests?

I can't find such tests
```
cd llvm
egrep -Irn "\-sycl\-std=((1\.2\.1)|(sycl\-1\.2\.1)|(121)|(2017))" . | egrep -v "^((\.git)|(\.vscode))" | wc -l
2

egrep -Irn "\-sycl\-std=((1\.2\.1)|(sycl\-1\.2\.1)|(121)|(2017)|(2020))" | egrep -v "^((\.git) | (\.vscode))" | wc -l
366
```

I'm not sure the old release notes from 13 months ago and from 4 ago have to be modified 
```
egrep -Irn "\-sycl\-std=((1\.2\.1)|(sycl\-1\.2\.1)|(121)|(2017))" | egrep -v "^((\.git)|(\.vscode))"
sycl/ReleaseNotes.md:749:- Deprecated usage of `-sycl-std=1.2.1` [8dd6fdff]
sycl/ReleaseNotes.md:4269:    they should be trivially copyable. The `-sycl-std=1.2.1` driver option turns
```

@dm-vodopyanov

**Comment by dm-vodopyanov** [comment]
> I can't find such tests

Looks like I have outdated branch in which these tests are present.

> I'm not sure the old release notes from 13 months ago and from 4 ago have to be modified

Yes, of course release notes should not be modified, they are not in my list above.

I cloned your repo to take a look closer: in the future, please don't create PR from `sycl` branch in your fork - create a branch with separate name in your fork.

Also this should be removed based on the PR title:

https://github.com/intel/llvm/blob/64a13200c43998043b6e240ab42bdc8e72982d26/clang/lib/Driver/ToolChains/Clang.cpp#L5512-L5521

https://github.com/intel/llvm/blob/64a13200c43998043b6e240ab42bdc8e72982d26/clang/test/Driver/sycl-deprecated.cpp#L10-L22

**Review by dm-vodopyanov — APPROVED** [review]

**Review by AlexeySachkov — APPROVED** [review]

**Review by mdtoguchi — APPROVED** [review]

**Review by smanna12 — APPROVED** [review]

**Comment by AndreiZibrov** [comment]
> I cloned your repo to take a look closer: in the future, please don't create PR from `sycl` branch in your fork - create a branch with separate name in your fork.

A misunderstanding occurred due to the simultaneous use of different branches😅. I temporarily set an incorrect branch as the default in my fork, which did not correspond to the branch used for the #14544.

As a result, an incorrect branch was utilized as default, leading to a discrepancy between the cloned files and the current state of the PR.

For precision, I would recommend using a specific commit ID or the PR ID as the target to be checked out as the end of the cloning process. This approach is detailed in the following resource: https://stackoverflow.com/questions/27567846/how-can-i-check-out-a-github-pull-request-with-git

I trust this clarifies the situation 😄

**Pull request merged**